cheroot/__main__.py

```diff
@@ -0,0 +1,161 @@
+"""Command line tool for starting a Cheroot WSGI instance.
+
+Basic usage::
+
+    # Start a server on 127.0.0.1:8000 with the default settings
+    # for the WSGI app myapp/wsgi.py:application()
+    cheroot myapp.wsgi
+
+    # Start a server on 0.0.0.0:9000 with 8 threads
+    # for the WSGI app myapp/wsgi.py:main_app()
+    cheroot myapp.wsgi:main_app --bind 0.0.0.0:9000 --threads 8
+
+    # Start a server on the UNIX socket /var/spool/myapp.sock
+    cheroot myapp.wsgi --bind /var/spool/myapp.sock
+
+    # Start a server on the abstract UNIX socket CherootServer
+    cheroot myapp.wsgi --bind @CherootServer
+"""
+
+import argparse
+from importlib import import_module
+import os
+import sys
+
+import six
+
+from .wsgi import Server
+
+
+def get_wsgi_app(wsgi_app_path):
+    """Read WSGI app path string and import application module."""
+    if ':' in wsgi_app_path:
+        mod_path, app_path = wsgi_app_path.split(':', 1)
+        if ':' in app_path:
+            raise ValueError(
+                'Application object name is invalid: it should not contain '
+                'colon (:) character.'
+            )
+    else:
+        mod_path, app_path = wsgi_app_path, 'application'
+
+    try:
+        wsgi_app = getattr(import_module(mod_path), app_path)
+        if not callable(wsgi_app):
+            raise TypeError
+    except ImportError as imp_err:
+        six.raise_from(NameError('Failed to find a module: {}'.format(
+            mod_path)), imp_err)
+    except AttributeError as attr_err:
+        six.raise_from(NameError(
+            'Failed to find application object: {}'.format(app_path)), attr_err
+        )
+    except TypeError as type_err:
+        six.raise_from(TypeError(
+            'Application must be a callable object'), type_err
+        )
+
+    return wsgi_app
+
+
+def parse_wsgi_bind_addr(bind_addr_string):
+    """Convert bind address string to a tuple."""
+    # try and match for an IP/hostname and port
+    match = six.moves.urllib.parse.urlparse('//{}'.format(bind_addr_string))
+    try:
+        addr = match.hostname
+        port = match.port
+        if (addr is not None) and (port is not None):
+            return addr, port
+    except ValueError:
+        pass
+
+    # else, assume a UNIX socket path
+    # if the string begins with an @ symbol, use an abstract socket
+    if bind_addr_string.startswith('@'):
+        return '\0{}'.format(bind_addr_string[1:]), None
+    return bind_addr_string, None
```

Comment:
Having plain tuple looks fragile. I'd prefer having dataclass for this or at least a `namedtuple`.